tests/rptest/tests/datalake/datalake_services.py

```diff
@@ -0,0 +1,143 @@
+
+
+class DatalakeServicesBase(IcebergRESTCatalogTest):
+    """All inclusive base class for implementing datalake tests. Includes the
+    boiler plate to enable datalake and bring up related services.
+    """
+    def __init__(self,
+                 test_ctx,
+                 filesystem_catalog_mode=True,
+                 *args,
+                 **kwargs):
+        super(DatalakeServicesBase,
+              self).__init__(test_ctx,
+                             extra_rp_conf={
+                                 "iceberg_enabled": "true",
+                                 "iceberg_catalog_commit_interval_ms": 5000
+                             },
+                             *args,
+                             **kwargs)
+        self.test_ctx = test_ctx
+        self.filesystem_catalog_mode = filesystem_catalog_mode
+        self.spark: Optional[SparkService] = None
+        self.trino: Optional[TrinoService] = None
+
+    def setUp(self):
+        self.catalog_service.set_filesystem_wrapper_mode(
+            self.filesystem_catalog_mode)
+        super().setUp()
+        si = self.redpanda.si_settings
+        self.spark = SparkService(
+            self.test_ctx,
+            iceberg_catalog_rest_uri=str(self.catalog_service.catalog_url),
+            cloud_storage_access_key=str(si.cloud_storage_access_key),
+            cloud_storage_secret_key=str(si.cloud_storage_secret_key),
+            cloud_storage_region=si.cloud_storage_region,
+            cloud_storage_api_endpoint=str(si.endpoint_url))
+        self.spark.start()
+        self.trino = TrinoService(
+            self.test_ctx,
+            iceberg_catalog_rest_uri=str(self.catalog_service.catalog_url),
+            cloud_storage_access_key=str(si.cloud_storage_access_key),
+            cloud_storage_secret_key=str(si.cloud_storage_secret_key),
+            cloud_storage_region=si.cloud_storage_region,
+            cloud_storage_api_endpoint=str(si.endpoint_url))
+        self.trino.start()
+
+    def tearDown(self):
+        if self.spark:
+            self.spark.stop()
+        if self.trino:
+            self.trino.stop()
+        return super().tearDown()
+
+    def create_iceberg_enabled_topic(self,
+                                     name,
+                                     partitions=1,
+                                     replicas=1,
+                                     translation_interval_ms=3000):
+        rpk = RpkTool(self.redpanda)
+        rpk.create_topic(topic=name,
+                         partitions=partitions,
+                         replicas=replicas,
+                         config={
+                             TopicSpec.PROPERTY_ICEBERG_ENABLED:
+                             "true",
+                             TopicSpec.PROPERTY_ICEBERG_TRANSLATION_INTERVAL:
+                             translation_interval_ms
+                         })
+
+    def wait_for_iceberg_table(self, schema, table, timeout, backoff_sec):
```

Comment:
Should `schema` be named `namespace`?

Reply:
Huh TIL, I guess that's true for Trino and Presto. I've only ever seen it refer to what Kafka and Iceberg refer to as schemas